lib/Reader/readerir.cpp

```diff
@@ -4099,18 +4105,24 @@ bool GenIR::canExpandMDArrayRef(CORINFO_SIG_INFO *Sig, bool IsStore,
   CORINFO_CLASS_HANDLE Class = nullptr;
   uint32_t ElemSize = 0;
   if (IsStore) {
-    CORINFO_ARG_LIST_HANDLE ArgLst = Sig->args;
+    CORINFO_ARG_LIST_HANDLE ArgList = Sig->args;
+    CORINFO_CLASS_HANDLE ArgType;
+
     // Skip arguments for each dimension.
     for (uint32_t R = 0; R < *Rank; R++) {
-      ArgLst = getArgNext(ArgLst);
+      CorType = strip(getArgType(Sig, ArgList, &ArgType));
+      if (CorType != CorInfoType::CORINFO_TYPE_INT) {
```

Comment:
It turns out that indices that are not int32s would be illegal w.r.t. ECMA, so I changed this to an assert.